pkg/common/configs/configvalidator.go

```diff
@@ -615,6 +723,13 @@ func Validate(newConfig *SchedulerConfig) error {
 		if err != nil {
 			return err
 		}
+
+		if err = checkLimitResource(partition.Queues[0], nil, make(map[string]*resources.Resource), make(map[string]*resources.Resource)); err != nil {
```

Comment:
If we can pass the init map, and when return err we clean the map.
When we don't return error, we can reuse the map to limit storage, we don't need to load from conifg to storage again after this. cc @wilfred-s @pbacsko

Reply:
We store limits containing max apps and max resources for users and groups in separate maps. But here, we handle only max resource and max applications in another method. I am not sure how we can club together and also not sure about the net gain. If really needed, can we file a follow up jira?

Reply:
Sure @manirajv06 , i think we may file a follow-up jira for this optimize